jiant/utils/python/datastructures.py

```diff
@@ -105,6 +105,12 @@ def sort_dict(d: dict):
     return {k: d[k] for k in sorted(list(d.keys()))}
 
 
+def set_dict_keys(d, key_list):
+    """Return a new dictionary with specified key order"""
+    assert set(d) == set(key_list)
+    return {k: d[k] for k in key_list}
```

Comment:
insertion order is only guaranteed in Python 3.6

Reply:
We're currently using dataclasses so we require at least 3.7.